backend/cn/check.ml

```diff
@@ -531,6 +530,26 @@ let rec check_pexpr (pe : BT.t mu_pexpr) (k : IT.t -> unit m) : unit m =
        | ty -> fail (fun _ -> {loc; msg = Mismatch {has = BT.pp ty; expect = !^"comparable type"}})
      in
      begin match op with
+     | OpAdd ->
+      let@ () = WellTyped.ensure_base_type loc ~expect: (bt_of_pexpr pe1) (bt_of_pexpr pe2) in
+         check_pexpr pe1 (fun v1 ->
+         check_pexpr pe2 (fun v2 ->
+         k (add_ (v1, v2) loc)))
+     | OpSub ->
+         let@ () = WellTyped.ensure_base_type loc ~expect: (bt_of_pexpr pe1) (bt_of_pexpr pe2) in
+         check_pexpr pe1 (fun v1 ->
+         check_pexpr pe2 (fun v2 ->
+         k (sub_ (v1, v2) loc)))
+     | OpMul ->
+         let@ () = WellTyped.ensure_base_type loc ~expect: (bt_of_pexpr pe1) (bt_of_pexpr pe2) in
+         check_pexpr pe1 (fun v1 ->
+         check_pexpr pe2 (fun v2 ->
+         k (mul_ (v1, v2) loc)))
+     | OpDiv ->
+         let@ () = WellTyped.ensure_base_type loc ~expect: (bt_of_pexpr pe1) (bt_of_pexpr pe2) in
+         check_pexpr pe1 (fun v1 ->
+         check_pexpr pe2 (fun v2 ->
+         k (div_ (v1, v2) loc)))
```

Comment:
I think this code is not needed and should be deleted (unless you have a test case that demonstrates otherwise).